tests/test_default_qubit.py

```diff
@@ -939,6 +968,47 @@ def circuit():
 
         assert np.allclose(circuit(), expected_output, atol=tol, rtol=0)
 
+    # This test is ran with two Z expvals
+    @pytest.mark.parametrize("name,par,wires,expected_output", [
+        ("BasisState", [1, 1], [0, 1], [-1, -1]),
+        ("BasisState", [1], [0], [-1, 1]),
+        ("BasisState", [1], [1], [1, -1])
+    ])
+    def test_basis_state_qubit_subset(self, qubit_device_2_wires, tol, name, par, wires, expected_output):
+        """Tests qubit basis state preparation on subsets of qubits"""
+
+        op = getattr(qml.ops, name)
+
+        @qml.qnode(qubit_device_2_wires)
+        def circuit():
+            op(np.array(par), wires=wires)
+            return qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))
+
+        assert np.allclose(circuit(), expected_output, atol=tol, rtol=0)
+
+    # This test is ran with two Z expvals
+    @pytest.mark.parametrize("name,par,wires,expected_output", [
+        ("QubitStateVector", [0, 1], [1], [1, -1]),
+        ("QubitStateVector", [0, 1], [0], [-1, 1]),
+        ("QubitStateVector", [1, 1], [1], [1, 0])
+    ])
+    def test_state_vector_qubit_subset(self, qubit_device_2_wires, tol, name, par, wires, expected_output):
+        """Tests qubit state vector preparation on subsets of qubits"""
+
+        op = getattr(qml.ops, name)
+
+        # normalize par
+        par = np.array(par)
+        par = par / np.linalg.norm(par, ord=2)
```

Comment:
I would refrain from hiding such things in the tests. Also because we use normalized state vectors in the whole rest of the tests. It would be better to parametrize the third test as `[1/np.sqrt(2), 1/np.sqrt(2)]`.

Reply:
I would also appreciate if you could add some tests with asymmetric and complex arguments like `[1j/2, np.sqrt(3)/2]`.